### ISSUE TITLE
Review Scala modeling and refactoring

### ISSUE THREAD

**Issue by angelcervera** [angelcervera/osm4scala#24]
For example, in WayEntity.scala it is used Enums and Java style hierarchy.
Two good explanation about why avoid Enums, at least in Scala:

- https://stackoverflow.com/questions/25305523/enum-vs-static-constants-memory-footprint
- https://stackoverflow.com/questions/1898932/case-objects-vs-enumerations-in-scala

**Comment by ericsun95** [comment]
I would prefer `ADT` or `seal trait` for OSMEntity and sub type using the case class.

**Comment by angelcervera** [comment]
Yes, agree 100% Enums is one of the main reasons to update it. Java mind from 2016 :smile:
